src/jukebox/components/synchronisation/sync_shared/__init__.py

```diff
@@ -0,0 +1,368 @@
+        if state == -1:
+            self.sync_card_database(card_id)
+
+    def _play_card_callback(self, folder: str, state: int):
+        if state == 0:
+            self.sync_folder(folder)
+
+    @plugs.tag
+    def sync_change_on_rfid_scan(self, option: str = 'toggle') -> None:
+        """
+        Change activation of 'on_rfid_scan_enabled'
+
+        :param option: Must be one of 'enable', 'disable', 'toggle'
+        """
+        if self._sync_enabled:
+
+            if option == 'enable':
+                _new_state = True
+            elif option == 'disable':
+                _new_state = False
+            elif option == 'toggle':
+                _new_state = not self._sync_on_rfid_scan_enabled
+            else:
+                logger.error(f"Invalid value '{option}' for 'option' in 'sync_change_on_rfid_scan'")
+                _new_state = None
+
+            if _new_state is not None:
+                cfg_sync_shared.setn('sync_shared', 'on_rfid_scan_enabled', value=_new_state)
+                self._sync_on_rfid_scan_enabled = _new_state
+
+                logger.info(f"Changed 'on_rfid_scan_enabled' to '{_new_state}'")
+
+        else:
+            logger.debug("Sync shared deactivated")
+
+    @plugs.tag
+    def sync_all(self) -> bool:
+        """
+        Sync all audiofolder and cardids from the remote server.
+        Removes local entries not existing at the remote server.
+        """
+        _files_synced = False
+
+        if self._precheck_enablement_sync():
+            logger.info("Syncing all")
+            _database_synced = self._sync_card_database()
+            _folder_synced = self._sync_folder('')
+            _files_synced = _database_synced or _folder_synced
+
+        return _files_synced
+
+    @plugs.tag
+    def sync_card_database(self, card_id: str) -> bool:
+        """
+        Sync the card database from the remote server, if existing.
+        If card_id is provided only this entry is updated.
+
+        :param card_id: The cardid to update
+        """
+        _files_synced = False
+
+        if self._precheck_enablement_sync_on_rfid_scan():
+            _files_synced = self._sync_card_database(card_id)
+
+        return _files_synced
+
+    @plugs.tag
+    def sync_folder(self, folder: str) -> bool:
+        """
+        Sync the folder from the remote server, if existing
+
+        :param folder: Folder path relative to music library path
+        """
+        _files_synced = False
+
+        if self._precheck_enablement_sync_on_rfid_scan():
+            _files_synced = self._sync_folder(folder)
+
+        return _files_synced
+
+    def _precheck_enablement_sync(self) -> bool:
+        if self._sync_enabled:
+            return True
+        else:
+            logger.debug("Sync shared deactivated")
+
+        return False
+
+    def _precheck_enablement_sync_on_rfid_scan(self) -> bool:
+        if self._precheck_enablement_sync():
+            if self._sync_on_rfid_scan_enabled:
```

Comment:
You could write this more concise

```python
if self.is_sync_enabled and self.sync_on_rfid_scan_enabled:
  return True

logger.debug("Sync on RFID scan deactivated")
return False
```

Reply:
For a more precise logging i would keep the seperation.
Otherwise if "is_sync_enabled" is false there would still be the log entry for "rfid scan" even if this setting might be activated, which could case confusion.